### ISSUE TITLE
Drop the Submodule for Bitcoin core and commit license directly

### ISSUE THREAD

**Issue by MattDHill** [Start9Labs/bitcoin-core-startos#83]

**Comment by MattDHill** [comment]
The only reason we are have the bitcoin submodule is to sync the license. Since Bitcoin's license isn't likely to ever change, this seems like a waste. We should just add an MIT license to the wrapper repo.

**Comment by chrisguida** [comment]
The license is updated every year to change the year: https://github.com/bitcoin/bitcoin/blob/master/COPYING

Are you saying we should copy this license file into the wrapper repo every year?

What's being wasted exactly? The submodule is merely a pointer to the upstream repo, takes up like 100 bytes. Also, the submodule points to a specific commit hash, which the Dockerfile does not do.

**Comment by MattDHill** [comment]
What does the commit hash have to do with this?

**Comment by MattDHill** [comment]
Useful for what. We only use it for a license that is MIT forever

**Comment by MattDHill** [comment]
Wait now I'm not seeing the submodule link to v22 anymore. Maybe things look different on mobile?

**Comment by chrisguida** [comment]
It's there, but you're right, it's pointing to v22 xD

<img width="982" alt="Screen Shot 2022-07-21 at 9 29 43 AM" src="https://user-images.githubusercontent.com/7445670/180239728-b846fe7b-9c89-4486-af03-ac13e51f8f45.png">

IMO we should just build from the git repo, that's what it's there for. Verifying all the PGP keys is sort of pointless and becoming harder to maintain.

CC @dr-bonez @kn0wmad

**Comment by chrisguida** [comment]
> Useful for what. We only use it for a license that is MIT forever

I just told you. It's useful because it points to a specific github commit...

**Comment by MattDHill** [comment]
I don't understand why having the commit hash for the license is important

**Comment by chrisguida** [comment]
Having the commit hash for the *code* is important

**Comment by MattDHill** [comment]
I agree, but this ticket is about NOT using the submodule for the license under the assumption we are using bitcoin.org for the code.

**Comment by chrisguida** [comment]
Right, and i'm arguing that, instead of implementing this ticket, we should instead switch away from building from bitcoin.org and build from the submodule instead. This is much more auditable and easy to maintain IMO. It's how we do it in pretty much all our other packages.

**Comment by MattDHill** [comment]
IDK @dr-bonez said he preferred it this way

**Comment by dr-bonez** [comment]
Who maintains the mirror on github for bitcoin core?

**Comment by dr-bonez** [comment]
Or is it the official repo?

**Comment by chrisguida** [comment]
Github is where all the dev happens... bitcoin.org is where the binaries are hosted. It makes sense to verify the keys when you're just installing a binary, but we're building from source, from the official repo. Let's just actually use the repo.
